### DISCUSSION TITLE
I do not understand this return fetch function

### DISCUSSION THREAD

**Discussion by DavidEdwardDarden** [nss-day-cohort-47/C47-Discussions#32] [Discuss a topic]
const getinventors = () => {

return fetch("https://gist.githubusercontent.com/brendalong/ab2424b0069ed0dd56b01951462a109d/raw/716c0776c45adea089644cffa7eadab139f9f47c/womensday.json")
.then(response => response.json())
.then(parsedResponse => {
    //return parsedResponse;
    console.log(parsedResponse);
})
}

getinventors();

I do not understand: 
Body.json() Takes a Response stream and reads it to completion. It returns a promise that resolves with the result of parsing the body text as JSON, which is a JavaScript value of datatype object, string, etc.

Googled: 
https://www.google.com/search?sxsrf=ALeKk03-Pl10JpWCTD6KhE3uZOyPCCzHmA%3A1615229184880&ei=AHFGYLufNfWYwbkP3uykwAg&q=.then+js&oq=.then+&gs_lcp=Cgdnd3Mtd2l6EAEYADIHCAAQhwIQFDICCAAyAggAMgIIADICCAAyAggAMgIIADICCAAyAggAMgIIADoJCAAQsAMQBxAeOgUIABCwA1CuFFiuFGDqImgBcAB4AIABW4gBrAGSAQEymAEAoAEBqgEHZ3dzLXdpesgBCsABAQ&sclient=gws-wiz
https://developer.mozilla.org/en-US/docs/Web/API/Response

**Comment by BryanNilsen** [comment]
The syntax that you have above can be broken down into repeatable concepts.
I suggest, at this point, trying to familiarize yourself with the bigger picture first before diving into the specifics of the .json() method:

1. `fetch("some_url_path_to_data")` - This is our GET **request** to where the data lives on some server. It takes an unknown time to get the data, so it returns a _promise_. Basically saying "I promise to return this information as soon as I can". So this code runs asynchronously while the rest of your code continues to run. In order to use the data when it comes back to us, we can use the `.then()` syntax to wait and **then** do some work when the data is available.

2. `.then(response => response.json())` -  After we make our request for data and it comes back to us, **then** we need to pull the data from the response and convert it from a readable text stream into JSON format so we can use the data in our code. This is also an asynchronous operation, so we need to have this line of code immediately after our GET request.

3. After we've made our GET request and have converted it into JSON, **THEN** we can use the data in our application. In this case, we are just logging the data to the console. We can do anything we want with the data at this point within the curly braces as shown below.
``` 
.then(parsedResponse => {
     console.log(parsedResponse);
     return parsedResponse
     ;})
``` 

Hang in there... get the basic concepts and syntax down first so you can identify and repeat the pattern and have a general idea what's going on. **THEN** you can dig deeper into HOW specific methods are doing what they do.
